lib/_http_outgoing.js

```diff
@@ -487,7 +492,7 @@ OutgoingMessage.prototype.end = function(data, encoding, callback) {
     encoding = null;
   }
 
-  if (data && typeof data !== 'string' && !(data instanceof Buffer)) {
+  if (data && !(typeof data === 'string' || data instanceof Buffer)) {
```

Comment:
Would you like me to also change [this line](https://github.com/iojs/io.js/pull/605/files#diff-286202fdbdd74ede6f5f5334b6176b5cR416)?